src/modules/orders/orders.service.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { Errors } from 'src/common/errors';
+import { ResponseCartItemDto } from 'src/modules/cart/response-cart.dto';
+import { OrderResponseDTO } from 'src/modules/orders/dto/order-response.dto';
+import { Order } from 'src/modules/orders/entities/order.entity';
+import { ProductResponseDTO } from 'src/modules/products/dto/product-response.dto';
+import { User } from 'src/modules/users/user.entity';
+
+import { Product } from '../products/entities/product.entity';
+
+import { CreateBuyerOrderDTO } from './dto/create-buyer-order.dto';
+import { GetBuyerOrderDTO } from './dto/get-buyer-order.dto';
+import { OrderDTO } from './dto/order.dto';
+import { BuyerOrder } from './entities/buyer-order.entity';
+import { Status } from './entities/order-status.enum';
+
+@Injectable()
+export class OrdersService {
+  constructor(
+    @InjectRepository(Order)
+    private readonly orderRepository: Repository<Order>,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<ProductResponseDTO>,
+    @InjectRepository(BuyerOrder)
+    private readonly buyerOrderRepository: Repository<BuyerOrder>,
+  ) {}
+
+  async findByVendor(vendorId: string): Promise<OrderResponseDTO[]> {
+    try {
+      const vendor = await this.userRepository.findOne({
+        where: { id: vendorId },
+      });
+
+      if (!vendor) {
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const orders = await this.orderRepository.find({
+        where: { vendorId: vendorId },
+        relations: ['products'],
+        order: { createdAt: 'DESC' },
+      });
+
+      if (!orders.length) {
+        throw new NotFoundException(Errors.ORDERS_NOT_FOUND);
+      }
+
+      return orders.map((order) => new OrderResponseDTO(order));
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(
+        Errors.FAILED_TO_FETCH_ORDERS_BY_VENDOR,
+      );
+    }
+  }
+
+  async createBuyerOrder(
+    responseCartItemDto: ResponseCartItemDto[],
+  ): Promise<CreateBuyerOrderDTO> {
+    const [{ userId }]: ResponseCartItemDto[] = responseCartItemDto;
+
+    try {
+      const buyer = await this.userRepository.findOne({
+        where: { id: userId },
+      });
+
+      if (!buyer) {
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const ordersMap = new Map<string, OrderDTO>();
+
```

Comment:
Why do we need to get the items from dto? We can just take all this data from the cart table. There we can find all the products which the current user wants to buy.
I need just a function which gets userId and shipping cost and create in DB buyer_order using the data from the cart table. And also it should create a order for every vendor in the user`s cart. All the products should be connected to their orders, all the orders should be connected to their buyer_order.

Reply:
done